### DISCUSSION TITLE
Why is WindTerm only "partially" open source?

### DISCUSSION THREAD

**Discussion by christianparpart** [kingToolbox/WindTerm#830] [Q&A]
Hey guys,

> Important: WindTerm is a partial open source project, and the source will be gradually opened.

this sounds like an interesting project. I am surprised I have never heard of you until now. My main concern is, while claiming being open source all of that being Apache-2.0, and all that being nice, why only partially? 

Thinking out loud, to me this sounds like you want to hide something. No offense. :-) But the provided source code is really useless and not even remotely giving an idea of what WindTerm is about behind the scenes. It simply feels wrong.

I'm not trying to force you to open up, but rather would like to understand your point of view and especially reasoning here.

**Comment by christianparpart** [comment]
@kingToolbox maybe you accidentally missed my question, so I'm pinging you, since you are indeed active in other posts in your project. :)

**Reply by sitemapxml** [reply]
I find this discussion quite interesting... [https://news.ycombinator.com/item?id=21230311](https://news.ycombinator.com/item?id=21230311)

**Reply by christianparpart** [reply]
Wait. What? @kingToolbox is actually active on HN but actively ignores my request for clarification here? That doesn't put the project into a good light, so it seems. Sad. 
p.s.: @sitemapxml thx for sharing the link. :)

**Reply by sitemapxml** [reply]
Yeah, it's kinda sad... For me, it is not a problem that this isn't open source, what I see as a problem is lack of transparency.
If this were a commercial product, I would be happy to buy a license for it, but promoting something as partially open source, and ignoring obvious questions puts many shades on this project.
@kingToolbox has mentioned somewhere that he has multiple unsuccessful projects which were open source, so that this time he doesn't want to make same mistakes as before, so it may be a reason for this suspicious behavior.
At least, we should be happy because discussions are enabled for this project 😃

**Reply by christianparpart** [reply]
late reply @sitemapxml, sorry. but do you have any reference to what you said? I mean, about the multiple failures he had in the past. What projects have that been? I am curious, actually. To me this sounds like the author is just trying to hide the source code as being afraid of competition that might potentially take some of the source code or implementation detail ideas to help the other people's projects. 

Providing a binary-only package, yet using Github to promote a project, to me, honestly, sounds like fake intent. So if the author continues like that, it'll just look like another failed project then (due to lack of transparency, no honest answers (not just to me), and the fake illusion to sell it as open source, yet only garbage (compared to what this project is actually like) is available here.

So yeah, that's my perception on this. So far I feel like I can only run this software within a sandbox, as I don't trust it. Why should I trust it? 🤔 

Take care!

**Reply by sitemapxml** [reply]
Hello Christian,
I have checked the link that I sent you earlier, and the part you are interested in is on the bottom of the page.
Here is a screenshot for the reference... https://i.imgur.com/v2F9kok.png
It is just one sentence, but quite enough to explain why the author is hiding source code.

> So far I feel like I can only run this software within a sandbox, as I don't trust it. Why should I trust it?

This make me laugh, because you have literally spoken my own words... :smile: <br>
It is interesting that there is a good deal of freeware software (mainly for Windows) that is not open source, but still is trusted and has many users. I would personally have a more trust to this project if it was just another freeware app hosted on its own website, than a freeware app hosted on GitHub. I think that choosing GitHub for hosting this project it actually the only wrong decision from the author, but still a major one...
